### PR TITLE
[Docs] Add Rollback instructions in the Upgrade docs

### PR DESCRIPTION
**Pull request by hari90**
Enhance the CoreDB upgrade docs and add instructions to Rollback the upgrade.

Same set of changes have been made to both Stable and Preview version of the docs.

### CODE REVIEW

File: docs/content/preview/manage/upgrade-deployment.md
Comment (lingamsandeep) [inline docs/content/preview/manage/upgrade-deployment.md:1]: Typo: I think you meant unusual behavior in a few places, but text says usual behavior. Even with the typo fixed, I think unusual sounds odd. How about - “If you encounter any issues after the upgrade, you have the option of Rolling back to the previous version”.

Comment (lingamsandeep) [reply docs/content/preview/manage/upgrade-deployment.md:1]: Rest of the document looks fine. Find all occurrences of “Unusual” or “Usual” due to typo and change.

**Review by lingamsandeep — APPROVED** [review]

File: docs/content/preview/manage/upgrade-deployment.md [hunk 1]
```diff
@@ -12,126 +12,158 @@ menu:
...
-## Upgrade the YSQL system catalog
+PromoteAutoFlags completed successfully
+No new AutoFlags eligible to promote
+Current config version: 1
+``````
 
-Similarly to PostgreSQL, YugabyteDB stores YSQL system metadata, referred to as the YSQL system catalog, in special tables. The metadata includes information about tables, columns, functions, users, and so on. The tables are stored separately, one for each database in the cluster.
+{{< note title="Note" >}}
+- `promote_auto_flags` is idempotent and can be run multiple times.
+- Before promoting AutoFlags, ensure that all YugabyteDB processes in the cluster have been upgraded to the new version. If any process running an old version may fail to connect to the cluster after the AutoFlags have been promoted.
```
Comment (ddhodge) [inline docs/content/preview/manage/upgrade-deployment.md:154]:
```suggestion
- Before promoting AutoFlags, ensure that all YugabyteDB processes in the cluster have been upgraded to the new version. Any process running an old version may fail to connect to the cluster after the AutoFlags have been promoted.
```

Comment (ddhodge) [inline docs/content/preview/manage/upgrade-deployment.md:154]: "YugabyteDB processes" - you mean masters and tservers? or something else?

File: docs/content/preview/manage/upgrade-deployment.md [hunk 2]
```diff
@@ -12,126 +12,158 @@ menu:
...
+{{< /note >}}
 
-When new features are added to YugabyteDB, objects such as new tables and functions need to be added to the system catalog. When you create a new cluster using the latest release, it is initialized with the most recent pre-packaged YSQL system catalog snapshot.
+2. Wait at least 10 seconds (`FLAGS_auto_flags_apply_delay_ms`) for the new AutoFlags to be propagated and applied on all YugabyteDB processes.
 
-However, the YugabyteDB upgrade process only upgrades binaries, and doesn't affect the YSQL system catalog of an existing cluster - it remains in the same state as before the upgrade. To derive the benefits of the latest YSQL features when upgrading, you need to manually upgrade the YSQL system catalog.
+#### 2. Upgrade the YSQL system catalog
+If you do not have [YSQL enabled](../../reference/configuration/yb-tserver/#ysql-flags), you can skip this step.
```
Comment (ddhodge) [inline docs/content/preview/manage/upgrade-deployment.md:160]:
```suggestion
If you do not have [YSQL enabled](../../reference/configuration/yb-tserver/#ysql), you can skip this step.
```

File: docs/content/preview/manage/upgrade-deployment.md [hunk 3]
```diff
@@ -145,54 +177,80 @@ Expect to see the following output:
 YSQL successfully upgraded to the latest version
 ```
 
-In certain scenarios, a YSQL upgrade can take longer than 60 seconds, which is the default timeout value for `yb-admin`. If this happens, run the following command with a greater timeout value:
+In certain scenarios, a YSQL upgrade can take longer than 60 seconds, which is the default timeout value for `yb-admin`. If this happens, run the command with a greater `-timeout_ms` value:
```
Comment (ddhodge) [inline docs/content/preview/manage/upgrade-deployment.md:180]:
```suggestion
In certain scenarios, a YSQL upgrade can take longer than 60 seconds, which is the default timeout value for `yb-admin`. If this happens, run the command with a greater `-timeout_ms` value. For example:
```

File: docs/content/preview/manage/upgrade-deployment.md [hunk 4]
```diff
@@ -145,54 +177,80 @@ Expect to see the following output:
...
 
+**Example**
```
Comment (ddhodge) [inline docs/content/preview/manage/upgrade-deployment.md:182]:
```suggestion
```

File: docs/content/preview/manage/upgrade-deployment.md [hunk 5]
```diff
@@ -145,54 +177,80 @@ Expect to see the following output:
...
+{{< note title="Note" >}}
+- `upgrade_ysql` is idempotent and can be run multiple times without any side effects.
+- Concurrent operations in a cluster can lead to transactional conflicts, catalog version mismatches, and read restart errors. This is expected, and should be addressed by retrying the operation. If `upgrade_ysql` encounters these errors, then it should also be retried.
+{{< /note >}}
+
+### B. Rollback Phase
 
-Concurrent operations in a cluster can lead to transactional conflicts, catalog version mismatches, and read restart errors. This is expected, and should be addressed by rerunning `upgrade_ysql`.
+{{< note title="Note" >}}
+- You cannot rollback after finalizing the upgrade.
```
Comment (ddhodge) [inline docs/content/preview/manage/upgrade-deployment.md:198]:
```suggestion
- You cannot perform a rollback after finalizing the upgrade.
```

File: docs/content/preview/manage/upgrade-deployment.md [hunk 6]
```diff
@@ -145,54 +177,80 @@ Expect to see the following output:
...
+- Rollback is only supported when you are upgrading a cluster that is already on version `2.20.2.0` or higher.
```
Comment (ddhodge) [inline docs/content/preview/manage/upgrade-deployment.md:199]:
```suggestion
- Rollback is only supported when you are upgrading a cluster that is on v2.20.2.0 or later.
```

File: docs/content/preview/manage/upgrade-deployment.md [hunk 7]
```diff
@@ -145,54 +177,80 @@ Expect to see the following output:
...
+{{< /note >}}
 
-## Upgrades and xCluster
+In order to rollback to the version that you were on before the upgrade, you need to restart all YB-Master and YB-Tservers on the old version.
```
Comment (ddhodge) [inline docs/content/preview/manage/upgrade-deployment.md:202]:
```suggestion
To roll back to the version that you were on before the upgrade, you need to restart all YB-Master and YB-TServers on the old version.
```

File: docs/content/preview/manage/upgrade-deployment.md [hunk 8]
```diff
@@ -145,54 +177,80 @@ Expect to see the following output:
...
 
-When xCluster replication is configured, replication needs to be temporarily paused when upgrading any of the clusters involved in xCluster replication.
+#### 1. Rollback YB-Tservers
```
Comment (ddhodge) [inline docs/content/preview/manage/upgrade-deployment.md:204]:
```suggestion
#### 1. Roll back YB-TServers
```

File: docs/content/preview/manage/upgrade-deployment.md [hunk 9]
```diff
@@ -145,54 +177,80 @@ Expect to see the following output:
...
 
-Use the following procedure to upgrade clusters involved in xCluster replication:
+Rollback the YB-Tservers one node at a time:
```
Comment (ddhodge) [inline docs/content/preview/manage/upgrade-deployment.md:206]:
```suggestion
Roll back the YB-TServers one node at a time:
```

File: docs/content/preview/manage/upgrade-deployment.md [hunk 10]
```diff
@@ -145,54 +177,80 @@ Expect to see the following output:
...
-    Replication disabled successfully
+    ```sh
+    cd /home/yugabyte/softwareyb-$OLD_VER/
     ```
 
-2. Proceed to perform upgrade of all the clusters involved.
-3. Resume replication on all the clusters involved using yb-admin.
+1. Start the old version of the YB-TServer process. Follow the instructions from [Start YB-TServers](../../deploy/manual-deployment/start-tservers/).
+
+1. Make sure that all YB-Tserver processes are running and the cluster load is balanced at `http://<any-yb-master>:7000/tablet-servers`.
```
Comment (ddhodge) [inline docs/content/preview/manage/upgrade-deployment.md:222]:
```suggestion
1. Make sure that all YB-TServer processes are running and the cluster load is balanced at `http://<any-yb-master>:7000/tablet-servers`.
```

File: docs/content/preview/manage/upgrade-deployment.md [hunk 11]
```diff
@@ -145,54 +177,80 @@ Expect to see the following output:
...
+
+1. Pause for at least 60 seconds before rolling back the next YB-TServer.
+
+#### 2. Rollback YB-Masters
```
Comment (ddhodge) [inline docs/content/preview/manage/upgrade-deployment.md:226]:
```suggestion
#### 2. Roll back YB-Masters
```

File: docs/content/preview/manage/upgrade-deployment.md [hunk 12]
```diff
@@ -145,54 +177,80 @@ Expect to see the following output:
...
+
+Use the following procedure to rollback all YB-Masters:
```
Comment (ddhodge) [inline docs/content/preview/manage/upgrade-deployment.md:228]:
```suggestion
Use the following procedure to roll back all YB-Masters:
```

File: docs/content/preview/manage/upgrade-deployment.md [hunk 13]
```diff
@@ -145,54 +177,80 @@ Expect to see the following output:
...
-    ```output
-    Replication enabled successfully
+    ```sh
+    cd /home/yugabyte/softwareyb-$OLD_VER/
     ```
 
-Downgrades are not currently supported. This is tracked in GitHub issue [#13686](https://github.com/yugabyte/yugabyte-db/issues/13686).
+1. Start the old version of the YB-Master process. Follow the instructions from [Start YB-Masters](../../deploy/manual-deployment/start-masters/).
+
+1. Make sure that all YB-Master processes are running at http://<any-yb-master>:7000/.
```
Comment (ddhodge) [inline docs/content/preview/manage/upgrade-deployment.md:244]:
```suggestion
1. Make sure that all YB-Master processes are running at `http://<any-yb-master>:7000/`.
```

File: docs/content/preview/manage/upgrade-deployment.md [hunk 14]
```diff
@@ -145,54 +177,80 @@ Expect to see the following output:
...
+
+1. Pause for at least 60 seconds before rolling back the next YB-Master.
+
+## Upgrades with xCluster
+
+When you have unidirectional xCluster replication, it is recommended to upgrade the target cluster before the source. Once the target cluster is upgraded and finalized, you can proceed to upgrade the source cluster.
+
+If you have bidirectional xCluster replication, then you should upgrade and finalize both clusters at the same time. Run the [Upgrade Phase](#upgrade-phase) for each cluster individually and monitor both of them. If you encounter any unusual behavior, rollback both clusters. If everything looks healthy, finalize both with as little delay as possible.
```
Comment (ddhodge) [inline docs/content/preview/manage/upgrade-deployment.md:252]:
```suggestion
If you have bidirectional xCluster replication, then you should upgrade and finalize both clusters at the same time. Run the [Upgrade Phase](#upgrade-phase) for each cluster individually and monitor both of them. If you encounter any issues, roll back both clusters. If everything looks healthy, finalize both with as little delay as possible.
```

File: docs/content/preview/manage/upgrade-deployment.md [hunk 15]
```diff
@@ -145,54 +177,80 @@ Expect to see the following output:
...
+
+{{< note title="Note" >}}
+xCluster replication requires the target cluster version to be equal to or higher than the source cluster version. The setup of a new xCluster replication will fail if this check fails. Existing replications will automatically pause if the source is finalized before the target.
```
Comment (ddhodge) [inline docs/content/preview/manage/upgrade-deployment.md:255]:
```suggestion
xCluster replication requires the target cluster version to be the same or later than the source cluster version. The setup of a new xCluster replication will fail if this check fails. Existing replications will automatically pause if the source is finalized before the target.
```

File: docs/content/stable/manage/upgrade-deployment.md [hunk 16]
```diff
@@ -1,137 +1,169 @@
...
-linkTitle: Upgrade a deployment
-description: Upgrade a deployment
+title: Upgrade YugabyteDB
+headerTitle: Upgrade YugabyteDB
+linkTitle: Upgrade YugabyteDB
+description: Upgrade YugabyteDB
 headcontent: Upgrade YugabyteDB on your deployment
 menu:
-  stable:
+  preview:
```
Comment (ddhodge) [inline docs/content/stable/manage/upgrade-deployment.md:8]:
```suggestion
  stable:
```

Comment (ddhodge) [inline docs/content/stable/manage/upgrade-deployment.md:8]: Presumably everything else is the same, and edits in preview can be copied to stable - however this line needs to stay set to 'stable'

**Review by ddhodge — APPROVED** [review]

File: docs/content/preview/manage/upgrade-deployment.md [hunk 17]
```diff
@@ -188,18 +187,18 @@ In certain scenarios, a YSQL upgrade can take longer than 60 seconds, which is t
...
-{{< note title="Note" >}}
-- You cannot rollback after finalizing the upgrade.
+{{< warning title="Important" >}}
 - Rollback is only supported when you are upgrading a cluster that is already on version `2.20.2.0` or higher.
-{{< /note >}}
+- You cannot rollback after finalizing the upgrade. If you still want to go back to the old version, you can [Export and import](../backup-restore/export-import-data/) the data to another cluster that is running the old version. You may have to manually modify the SQL script to make it match the old version's format.
+{{< /warning >}}
 
-In order to rollback to the version that you were on before the upgrade, you need to restart all YB-Master and YB-Tservers on the old version.
+In order to rollback to the version that you were on before the upgrade, you need to restart all YB-Master and YB-Tservers on the old version. All YB-Tservers have to be rolled back before you rollback YB-Masters.
```
Comment (ddhodge) [inline docs/content/preview/manage/upgrade-deployment.md]:
```suggestion
In order to roll back to the version that you were on before the upgrade, you need to restart all YB-Master and YB-TServers on the old version. All YB-TServers have to be rolled back before you roll back YB-Masters.
```

**Pull request merged**